### PR TITLE
Move RearmTick to Rearmable

### PR DESCRIPTION
**Pull request by PunkPun**
Split from #20168

### CODE REVIEW

File: OpenRA.Mods.Common/Traits/Rearmable.cs [hunk 1]
```diff
@@ -56,5 +58,31 @@ void INotifyResupply.BeforeResupply(Actor self, Actor target, ResupplyType types
...
+					}
+
+					rearmComplete = false;
+				}
+			}
+
+			if (rearmComplete)
+				return true;
+
+			return false;
```
Comment (Mailaender) [inline OpenRA.Mods.Common/Traits/Rearmable.cs]:
```suggestion
				return rearmComplete;
```

**Comment by Mailaender** [comment]
Can you explain why this change is an improvement?

**Comment by PunkPun** [comment]
`Resupply` activity will get removed and replaced by `DockActivity` and `DockSequence` activities, `RearmTick` is the only thing that matters to us

**Review by Mailaender — DISMISSED** [review]

**Comment by penev92** [comment]
Needs a rebase.

**Comment by PunkPun** [comment]
Rebased

**Review by penev92 — APPROVED** [review]

**Comment by penev92** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/4057ad57cddeea418060b87ff0133a351a79aac8...1cd13a6e3b0d3f258483ba154af6962856106cc4)

**Pull request merged**